src/common/snippets/include/snippets/op/brgemm.hpp

```diff
@@ -17,33 +17,51 @@ namespace op {
  * @brief Brgemm is a batch-reduced matrix multiplication with the support of arbitrary strides between matrices rows
  * @ingroup snippets
  */
-class Brgemm : public MemoryAccess {
+class Brgemm : virtual public modifier::MemoryAccess, public ov::op::Op {
 public:
-    OPENVINO_OP("Brgemm", "SnippetsOpset", MemoryAccess);
+    OPENVINO_OP("Brgemm", "SnippetsOpset");
     Brgemm(const Output<Node>& A, const Output<Node>& B,
            const size_t offset_a = 0lu, const size_t offset_b = 0lu, const size_t offset_c = 0lu,
-           std::vector<size_t> layout_a = {}, std::vector<size_t> layout_b = {}, std::vector<size_t> layout_c = {});
+           std::vector<size_t> layout_a = {}, std::vector<size_t> layout_b = {}, std::vector<size_t> layout_c = {},
+           size_t blk_size_m = 0, size_t blk_size_k = 0, size_t blk_size_n = 0);
     Brgemm(const Output<Node>& A, const Output<Node>& B,
            const PortDescriptor& desc_a, const PortDescriptor& desc_b, const PortDescriptor& desc_c,
-           std::vector<size_t> layout_a = {}, std::vector<size_t> layout_b = {}, std::vector<size_t> layout_c = {});
+           std::vector<size_t> layout_a = {}, std::vector<size_t> layout_b = {}, std::vector<size_t> layout_c = {},
+           size_t blk_size_m = 0, size_t blk_size_k = 0, size_t blk_size_n = 0);
     Brgemm() = default;
 
     size_t get_offset_a() const { return get_input_offset(0); }
     size_t get_offset_b() const { return get_input_offset(1); }
     size_t get_offset_c() const { return get_output_offset(0); }
 
+    size_t get_m_block_size() const { return m_M_blk; }
+    size_t get_k_block_size() const { return m_K_blk; }
+    size_t get_n_block_size() const { return m_N_blk; }
+    float get_beta() const { return m_beta; }
+
```

Comment:
It's really should be in Snippets `Brgemm`?

Reply:
Yep, I think so.
The thing is that both `BrgemmTPP` and `BrgemmCPU` are [derived from](https://github.com/IvanNovoselov/openvino/blob/92e48bd9a7d8a55c005c380d8dbb60c55b5ced71/src/plugins/intel_cpu/src/transformations/snippets/tpp/op/brgemm.hpp#L21) `snippets::op::Brgemm`. Blocking is a shared functionality of Brgemm, that is implemented on snippets (not backend) side, so it makes sense to move it to the base class.